### PR TITLE
Set edition to 2021 and run clippy

### PR DESCRIPTION
**Pull request by jcarres-mdsol**
I was curious. Changes:
- change edition to 2021
- run `cargo update` 
- run `cargo clippy` and do the fixes 
- run `cargo build` and `cargo test`

Everything seems fine. @masongup-mdsol

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 name = "mauth-client"
 version = "0.1.0"
```
Comment (masongup-mdsol) [inline Cargo.toml:3]: Can you bump the version as well, so I can push the updated version to crates.io?

**Review by masongup-mdsol — APPROVED** [review]
Looks good to me

File: Cargo.toml (hunk 1 above)
Comment (jcarres-mdsol) [reply Cargo.toml:3]: I thought of that but then, there are no changes really, I think not worth a version. This is mostly so it does not bitrot and when features come is in a good state

Comment (masongup-mdsol) [reply Cargo.toml:3]: That sounds right actually when I checked again how Rust Editions actually work 👍

**Pull request merged**